Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM rust:1.69-bullseye as rust-build-stage
+RUN apt-get update && apt-get upgrade -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /build
+COPY Cargo.lock Cargo.toml ./
+COPY src src/
+COPY config config/
+COPY themes themes/
+RUN cargo build --release
+
+FROM gcr.io/distroless/cc-debian10
+USER 1001:1001
+
+WORKDIR /app
+COPY --from=rust-build-stage --chown=1001:1001 /build/target/release/vivid .
```

Comment:
So... I think this will always use the builtin/embedded filetypes database and themes. So there is no way for the user to customize this using `~/.config/vivid/filetypes.yml` or `/usr/share/vivid/filetypes.yml`, for example.

Reply:
The goal would be for the `vivid` process to have access to these files at runtime, I think. That is, we wouldn't want to have the user rebuild the container locally every time they update their local configuration, right?

Reply:
Correct. So, the user would have to `--mount` their host ~/.config/vivid folder (and others)?

Reply:
For context, my use case is that I co-maintain [`LS_COLORS`](https://github.com/trapd00r/LS_COLORS). The vast number of extensions we support is getting unwieldy, so I'm interested in migrating our build to leverage `vivid`. Having vivid in docker would help a lot since that obviates managing a hermetic rust build process inside GHA. For a use case like that, I think it's perfectly fine to leave configuration and filesystem access to the user. Anybody who is used to working with docker is already used to dealing with that, and frankly, I for one prefer it that way.

Using docker to run applications on a consumer desktop system is not something I generally recommend, since the daemon runs as the root user and therefore it is unsafe in multi-user environments.

Reply:
> For context, my use case is that I co-maintain [`LS_COLORS`](https://github.com/trapd00r/LS_COLORS).

Oh, cool! That project was a huge inspiration for `vivid`. Glad to see that it is still actively maintained.

> The vast number of extensions we support is getting unwieldy, so I'm interested in migrating our build to leverage vivid.

That would be fantastic. We would certainly see some synergy effects in both projects.

> Having vivid in docker would help a lot since that obviates managing a hermetic rust build process inside GHA. For a use case like that, I think it's perfectly fine to leave configuration and filesystem access to the user.

I understand.

> Using docker to run applications on a consumer desktop system is not something I generally recommend, since the daemon runs as the root user and therefore it is unsafe in multi-user environments.

Ok, then let's leave the setup minimal and add the Dockerfile just for purposes like yours.